packages/flutter/lib/src/material/color_scheme.dart

```diff
@@ -97,6 +145,127 @@ class ColorScheme with Diagnosticable {
        _primaryVariant = primaryVariant,
        _secondaryVariant = secondaryVariant;
 
+  /// Generate a [ColorScheme] derived from the given `seedColor`.
+  ///
+  /// Using the seedColor as a starting point, a set of tonal palettes are
+  /// constructed. These tonal palettes are based on the Material 3 Color
+  /// system and provide all the needed colors for a [ColorScheme]. These
+  /// colors are designed to work well together and meet contrast
+  /// requirements for accessibility.
+  ///
+  /// If any of the optional color parameters are non-null they will be
+  /// used in place of the generated colors for that field in the resulting
+  /// color scheme. This allows apps to override specific colors for their
+  /// needs.
+  ///
+  /// Given the nature of the algorithm, the seedColor may not wind up as
+  /// one of the ColorScheme colors.
+  ///
+  /// See also:
+  ///
+  ///  * <https://m3.material.io/styles/color/the-color-system/color-roles>, the
+  ///    Material 3 Color system specification.
+  ///  * <https://pub.dev/packages/material_color_utilities>, the package
+  ///    used to generate the tonal palettes needed for the scheme.
+  factory ColorScheme.fromSeed({
+    required Color seedColor,
+    Brightness brightness = Brightness.light,
+    Color? primary,
+    Color? onPrimary,
+    Color? primaryContainer,
+    Color? onPrimaryContainer,
+    Color? secondary,
+    Color? onSecondary,
+    Color? secondaryContainer,
+    Color? onSecondaryContainer,
+    Color? tertiary,
+    Color? onTertiary,
+    Color? tertiaryContainer,
+    Color? onTertiaryContainer,
+    Color? error,
+    Color? onError,
+    Color? errorContainer,
+    Color? onErrorContainer,
+    Color? outline,
+    Color? background,
+    Color? onBackground,
+    Color? surface,
+    Color? onSurface,
+    Color? surfaceVariant,
+    Color? onSurfaceVariant,
+    Color? inverseSurface,
+    Color? onInverseSurface,
+    Color? inversePrimary,
+    Color? shadow,
+  }) {
+    final CorePalette palette = CorePalette.of(seedColor.value);
+    switch (brightness) {
+      case Brightness.light:
+        return ColorScheme(
+          primary: primary ?? Color(palette.primary.get(40)),
```

Comment:
Since these numerical values are most likely to change (just did last week), I think it'd be idea for use to use `Scheme` and create the `ColorScheme` instance indirectly, etc. `primary: scheme.primary`. That way, no need to keep it in sync manually